code/datastructure-set/src/main/java/ru/unn/agile/datastructure/set/model/MySet.java

```diff
@@ -0,0 +1,97 @@
+package ru.unn.agile.datastructure.set.model;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+
+public class MySet<E> {
+
+    private static final String ERROR_MESSAGE = "MySet collection doesn't support Null values";
+    private List<E> container;
+
+    public MySet() {
+        container = new ArrayList<>();
+    }
+
+    public int size() {
+        return this.container.size();
+    }
+
+    public boolean add(final E e) {
+        if (e == null) {
+            throw new NullPointerException(ERROR_MESSAGE);
```

Comment:
Maybe it would be better to throw IllegalArgumentException?

Reply:
It will be better if it was validation for this goal. But in this case I'm checking the null value.
and other possible values couldn't be compared. Runtime checking will prevent this.

Reply:
Fixed. Discussed internally. I've got the point.